server/backend/mlx_engine/utils/progress_decorators.py

```
@@ -0,0 +1,100 @@
+
+    return inner_callback
+
+
+def token_count(
+    callback: Optional[Callable[[float], bool]],
+) -> Optional[Callable[[int, int], None]]:
+    """
+    Adapts a float percentage based progress callback into a token count based one.
+
+    Args:
+        outer_callback: A callback that accepts progress (0.0–100.0) and returns
```

Ruff (0.14.11)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)